### PR TITLE
Manual cherry pick of #89833: preserve integers decoding raw JSON values

### PR DESCRIPTION
**Pull request by liggitt**
Cherry pick of #89833 on release-1.17.

#89833: preserve integers decoding raw JSON values

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

**Does this PR introduce a user-facing change?**:
```release-note
Fixes a bug defining a default value for a replicas field in a custom resource definition that has the scale subresource enabled
```

/kind bug
/sig api-machinery
/priority important-soon
/cc @sttts @apelisse

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -152,7 +152,7 @@ require (
 	k8s.io/klog v1.0.0
 	k8s.io/kube-aggregator v0.0.0
 	k8s.io/kube-controller-manager v0.0.0
-	k8s.io/kube-openapi v0.0.0-20200316234421-82d701f24f9d
+	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29 // release-1.17
```
Comment (sttts) [inline go.mod:155]: does go keep these comments now or even adds them autmatically?

Comment (liggitt) [reply go.mod:155]: I added them, go preserves them

**Pull request merged**